### PR TITLE
Add E2E JWT tests

### PR DESCRIPTION
**Pull request by testableapple**
## Changes
- Add E2E JWT tests
- Create new configuration `Test` to make it easier to use environment variables and macros across all test targets
- Moved `Credentials.xcconfig` to a shared place

## Required actions locally:
- Move `StreamVideoTests/IntegrationTests/Credentials.xcconfig` to `TestTools/Credentials.xcconfig`

### CODE REVIEW

File: SwiftUIDemoAppUITests/SwiftUIDemoApp.xctestplan [hunk 1]
```diff
@@ -9,7 +9,17 @@
...
   ],
   "defaultOptions" : {
-
+    "environmentVariableEntries" : [
+      {
+        "key" : "STREAM_VIDEO_SECRET",
+        "value" : "$(STREAM_VIDEO_SECRET)"
+      }
+    ],
+    "targetForVariableExpansion" : {
```
Comment (martinmitrevski) [inline SwiftUIDemoAppUITests/SwiftUIDemoApp.xctestplan:18]: What is this doing?

File: TestTools/TestData/TokenGenerator.swift [hunk 2]
```diff
@@ -29,7 +29,28 @@ class TokenGenerator {
     
     static let shared = TokenGenerator()
     
-    func fetchToken(for userId: String, expiration tokenDurationInMinutes: Double = 0) -> UserToken? {
+    var retries = 0
+    
+    func fetchToken(for userId: String, expiration tokenDurationInSeconds: Int = 0) -> UserToken? {
+        // When tokens expires after #{expiration} seconds, what to do?
```
Comment (martinmitrevski) [inline TestTools/TestData/TokenGenerator.swift:35]: Is this forgotten? If a token expires, the backend returns 401, while the SDK calls the token provider to refresh the token.

File: SwiftUIDemoAppUITests/Pages/CallDetailsPage.swift [hunk 3]
```diff
@@ -25,4 +25,8 @@ enum CallDetailsPage {
         }
     }
     static var participants: XCUIElementQuery { participantList.buttons }
+    static var connectionErrorAlert: XCUIElement {
+        app.staticTexts.matching(NSPredicate(format: "label CONTAINS 'StreamVideo.APIError'")).firstMatch
+
```
Comment (martinmitrevski) [inline SwiftUIDemoAppUITests/Pages/CallDetailsPage.swift:30]: extra new line here

**Review by martinmitrevski — APPROVED** [review]
nice work 👍  Few small things to check (see comments).

File: SwiftUIDemoAppUITests/SwiftUIDemoApp.xctestplan (hunk 1 above)
Comment (testableapple) [reply SwiftUIDemoAppUITests/SwiftUIDemoApp.xctestplan:18]: `STREAM_VIDEO_SECRET`? We need to pass this env var to tests so they can mock JWT.

File: TestTools/TestData/TokenGenerator.swift (hunk 2 above)
Comment (testableapple) [reply TestTools/TestData/TokenGenerator.swift]: Oops, these are development/debugging comments 😄
Deleted 👍

**Comment by testableapple** [comment]
As soon as I turned off the recording of the tests, JWT tests started failing. Investigatin'...

File: SwiftUIDemoAppUITests/SwiftUIDemoApp.xctestplan (hunk 1 above)
Comment (martinmitrevski) [reply SwiftUIDemoAppUITests/SwiftUIDemoApp.xctestplan:18]: no, I meant this "targetForVariableExpansion".

Comment (testableapple) [reply SwiftUIDemoAppUITests/SwiftUIDemoApp.xctestplan:18]: Oh, this means that this variable will be passed to the `DemoApp` target through `xctestplan`

Comment (testableapple) [reply SwiftUIDemoAppUITests/SwiftUIDemoApp.xctestplan:18]: The UI tests can interact only with the app, so if want the SDK to receive this env var, we need to pass it to the app. Unit tests can do this straight away.

**Comment by testableapple** [comment]
> As soon as I turned off the recording of the tests, JWT tests started failing. Investigatin'...

Fixed

**Pull request merged**